EIPS/eip-6260.md

```diff
@@ -0,0 +1,290 @@
+     */
+    event UpdateRoyalty(uint indexed royalty);
+
+
+    /**
+     * @notice Puts a token to sale and set its price.
+     * @dev Requirements:
+     *
+     * - Owner of `_tokenId` must be the caller
+     *
+     * Emits an {UpdatePrice} event.
+     *
+     * @param _tokenId uint representing the token ID number.
+     * @param _price uint representing the price at which to sell the token.
+     */
+    function setPrice(uint _tokenId, uint _price) external;
+
+    /**
+     * @notice Removes a token from the sale.
+     * @dev Requirements:
+     *
+     * - Owner of `_tokenId` must be the caller
+     *
+     * Emits a {RemoveFromSale} event.
+     *
+     * @param _tokenId uint representing the token ID number.
+     */
+    function removeTokenSale(uint _tokenId) external;
+
+    /**
+     * @notice Buys a specific token from its ID onchain.
+     * @dev Amount of ether msg.value sent is transferred to `seller` of the token.
+     * A percentage of the royalty allocution is sent to `_owner` of the contract.
+     * The token of ID `_tokenId` is then transferred from `seller` to `buyer` (the msg.sender).
+     * The token is then automatically removed from the sale.
+     *
+     * Requirements:
+     *
+     * - `_tokenId` must be put to sale
+     * - Amount of ether `msg.value` sent must be greater than the selling price
+     *
+     * Emits a {Purchase} event.
+     *
+     * @param _tokenId uint representing the token Id number.
+     */
+    function buyToken(uint _tokenId) external payable;
+
+
+    /**
+     * @notice Return the current royalty and its denominator.
+     * @dev Return the current royalty and its denominator.
+     * @return _royalty uint within the range of `_royaltyDenominator` associated with the token.
+     * @return _denominator uint denominator set in `_royaltyDenominator()`.
+     */
+    function royaltyInfo() external view returns(uint, uint);
+
+    /**
+     * @notice Set the royalty percentage.
+     * @dev Set or update the royalty percentage within the range of `_royaltyDenominator`.
+     * Update the `_firstRoyaltyUpdate` boolean to true if previously false.
+     *
+     * Requirements:
+     *
+     * - caller must be `_owner` of the contract
+     * - `_newRoyalty` must be between 0 and `_royaltyDenominator`
+     * - `_newRoyalty` must be lower than current previous one
+     *
+     * Emits an {UpdateRoyalty} event.
+     *
+     * @param _newRoyalty uint within the range of `_royaltyDenominator` as new tokens royalties.
+     */
+    function setRoyalty(uint _newRoyalty) external;
+}
+```
+
+To create and deploy this EIP contract one **MAY** only inherit directly from this [EIP-721](./eip-721.md) extension that would directly signal support for this EIP:
+
+```javascript
+pragma solidity ^0.8.0;
+import "./ERC721Buyable.sol";
+
+contract NFTContract is ERC721Buyable {
+    // ...
+    constructor() ERC721("NFTBuyableContract", "NFTBC") { }
+    // ...
+}
+```
```

Comment:
This whole block is an implementation detail, and shouldn't be in your proposal. Implementers are free to choose how they make a compatible contract.